File: 0155-min-stack/0155-min-stack.cpp
class MinStack {
public:
    stack<int> min_stack, stk;
    MinStack() {
        
    }
    
    void push(int val) {
        stk.push(val);
        val = min(val, min_stack.empty() ? val : min_stack.top());
        min_stack.push(val);
    }
    
    void pop() {
        stk.pop();
        min_stack.pop();
    }
    
    int top() {
        return stk.top();
    }
    
    int getMin() {
        return min_stack.top();
    }
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack* obj = new MinStack();
 * obj->push(val);
 * obj->pop();
 * int param_3 = obj->top();
 * int param_4 = obj->getMin();
 */

File: 0155-min-stack/0155-min-stack.py
class MinStack:

    def __init__(self):
        self.stk = []
        self.stk_min = []

    def push(self, val: int) -> None:
        self.stk.append(val)
        val = min(val, self.stk_min[-1] if self.stk_min else val)
        self.stk_min.append(val)
        
    def pop(self) -> None:
        self.stk.pop()
        self.stk_min.pop()

    def top(self) -> int:
        return self.stk[-1]

    def getMin(self) -> int:
        return self.stk_min[-1]
